### PR TITLE
implicit value classes

### PR DESCRIPTION
**Pull request by gmethvin**

### CODE REVIEW

File: shoebox/app/com/keepit/classify/DomainTag.scala [hunk 1]
```diff
@@ -23,11 +23,10 @@ case class DomainTag(
   def withState(state: State[DomainTag]) = this.copy(state = state)
 }
 
-sealed trait DomainTagName { def name: String }
-private final case class DomainTagNameImpl(name: String) extends DomainTagName
+class DomainTagName private (val name: String) extends AnyVal
```
Comment (eishay) [inline shoebox/app/com/keepit/classify/DomainTag.scala:26]: nice!

File: shoebox/app/com/keepit/common/net/package.scala [hunk 2]
```diff
@@ -8,7 +8,7 @@ import com.keepit.common.healthcheck.HealthcheckError
 
 package object net {
 
-  class RichRequestHeader(request: RequestHeader) {
+  implicit class RichRequestHeader(request: RequestHeader) {
```
Comment (eishay) [inline shoebox/app/com/keepit/common/net/package.scala:11]: cool

File: shoebox/app/com/keepit/common/time/package.scala [hunk 3]
```diff
@@ -85,22 +81,16 @@ package object time {
 
     def isSameDay(ld: LocalDate)(implicit zone: DateTimeZone): Boolean = ld == date.withZone(zone).toLocalDate
 
-    lazy val format = HTTP_HEADER_DATETIME_FORMAT.print(date)
+    def format = HTTP_HEADER_DATETIME_FORMAT.print(date)
```
Comment (2is10) [inline shoebox/app/com/keepit/common/time/package.scala:84]: why the change from `lazy val` to `def`?  `DateTime` is immutable.

Comment (gmethvin) [inline shoebox/app/com/keepit/common/time/package.scala:84]: Value classes can only have one field, so we can't use a `val`. Since this is an implicit wrapper, the method should only be called once for each "instance" of the class. I think a `def` is better since we don't have the extra overhead of the `lazy val`.

Comment (2is10) [inline shoebox/app/com/keepit/common/time/package.scala:84]: Got it. Thanks.
